js/src/main/scala/org/widok/validation/Validator.scala

```diff
@@ -0,0 +1,28 @@
+package org.widok.validation
+
+import org.widok.{Dict, ReadStateChannel}
+
+case class Validator(validationSources: Tuple2[ReadStateChannel[_], Seq[FieldValidation[_]]]*) {
+
+  val validations = Dict[ReadStateChannel[_], Seq[String]]()
+
+  val valid = validations.forall(_.isEmpty).cache(true)
+  val errors = validations.filter(_.nonEmpty).buffer
+
+  validationSources.foreach {
+    case (channel, fieldValidators) =>
+      // don't validate initial values (tail) and only validate when the value is actually updated (distinct)
+      channel.distinct.tail.attach(input => validateValue(channel, fieldValidators, input))
+  }
+
+  private def validateValue(channel: ReadStateChannel[_], fieldValidators: Seq[FieldValidation[_]], input: Any) = {
```

Comment:
I'd suggest to remove the ``=``, to make it obvious that the method doesn't return anything.

Would it be able to make the types more specific? Perhaps something along the lines:

```scala
private def validateValue[T](channel: ReadStateChannel[T], fieldValidators: Seq[FieldValidation[T]], input: T)
```